File: src/Application Specific/Classes/Entity Collections/UserLevel/UserLevelSQL.vb
Imports System.Data.SqlClient

Namespace Entity.UserLevels

    Public Class UserLevelSQL
        Private _database As Sys.Database

        Public Sub New(ByVal database As Sys.Database)
            _database = database
        End Sub

#Region "Functions"

        Public Sub [Insert](ByVal UserID As Integer, ByVal t As DataTable)

            'delete all first
            _database.ClearParameter()
            _database.AddParameter("@UserID", UserID, True)
            _database.ExecuteSP_NO_Return("UserLevel_Delete")

            If Not t Is Nothing Then
                For Each r As DataRow In t.Rows
                    _database.ClearParameter()
                    _database.AddParameter("@UserID", UserID, True)
                    _database.AddParameter("@LevelID", r("LevelID"), True)
                    _database.AddParameter("@Notes", Entity.Sys.Helper.MakeStringValid(r("Notes")), True)
                    _database.AddParameter("@DatePassed", r("DatePassed"), True)
                    _database.AddParameter("@DateExpires", r("DateExpires"), True)
                    _database.AddParameter("@DateBooked", r("DateBooked"), True)
                    _database.ExecuteSP_NO_Return("UserLevel_Insert")
                Next
            End If
        End Sub

        Public Function [Get](ByVal UserID As Integer) As DataView
            Dim command As New SqlCommand("UserLevels_Get", New SqlConnection(_database.ServerConnectionString))
            command.CommandType = CommandType.StoredProcedure
            command.Parameters.AddWithValue("@UserID", UserID)

            Dim dt As DataTable = _database.ExecuteCommand_DataTable(command)
            dt.TableName = Entity.Sys.Enums.TableNames.tblUserQualification.ToString
            Return New DataView(dt)
        End Function

        Public Function [GetForSetup](ByVal UserID As Integer) As DataView
            _database.ClearParameter()
            _database.AddParameter("@UserID", UserID, True)
            Dim dt As DataTable = _database.ExecuteSP_DataTable("UserLevels_Setup_Get")
            dt.TableName = Entity.Sys.Enums.TableNames.tblUserQualification.ToString
            Return New DataView(dt)
        End Function

#End Region

    End Class

End Namespace

File: src/Application Specific/Classes/Entity Collections/UserLevel/UserLevelSQL.cs
using System.Data;
using System.Data.SqlClient;

namespace FSM.Entity.UserLevels
{
    public class UserLevelSQL
    {
        private Sys.Database _database;

        public UserLevelSQL(Sys.Database database)
        {
            _database = database;
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        public void Insert(int UserID, DataTable t)
        {

            // delete all first
            _database.ClearParameter();
            _database.AddParameter("@UserID", UserID, true);
            _database.ExecuteSP_NO_Return("UserLevel_Delete");
            if (t is object)
            {
                foreach (DataRow r in t.Rows)
                {
                    _database.ClearParameter();
                    _database.AddParameter("@UserID", UserID, true);
                    _database.AddParameter("@LevelID", r["LevelID"], true);
                    _database.AddParameter("@Notes", Sys.Helper.MakeStringValid(r["Notes"]), true);
                    _database.AddParameter("@DatePassed", r["DatePassed"], true);
                    _database.AddParameter("@DateExpires", r["DateExpires"], true);
                    _database.AddParameter("@DateBooked", r["DateBooked"], true);
                    _database.ExecuteSP_NO_Return("UserLevel_Insert");
                }
            }
        }

        public DataView Get(int UserID)
        {
            var command = new SqlCommand("UserLevels_Get", new SqlConnection(_database.ServerConnectionString));
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.AddWithValue("@UserID", UserID);
            var dt = _database.ExecuteCommand_DataTable(command);
            dt.TableName = Sys.Enums.TableNames.tblUserQualification.ToString();
            return new DataView(dt);
        }

        public DataView GetForSetup(int UserID)
        {
            _database.ClearParameter();
            _database.AddParameter("@UserID", UserID, true);
            var dt = _database.ExecuteSP_DataTable("UserLevels_Setup_Get");
            dt.TableName = Sys.Enums.TableNames.tblUserQualification.ToString();
            return new DataView(dt);
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }
}